engine/src/utils/format.rs

```diff
@@ -16,23 +22,71 @@ where
         let width = self[0].as_ref().len() as u32;
         let height = self.len() as u32;
 
-        let data: Vec<u8> = self
-            .iter()
-            .flat_map(|slice| {
-                slice
-                    .as_ref()
+        let mut data = vec![];
+        {
+            let mut encoder = PngEncoder::new(&mut data);
+
+            encoder.write_image(
```

Comment:
The result of this isn't being handled (you can just call `expect`, just like on the `to_gif` fn).